### DISCUSSION TITLE
Updated to v0.1.12 and started getting errors from parsing .map files from dependencies under node_modules

### DISCUSSION THREAD

**Discussion by pushkargupta9807** [web-infra-dev/rspack#3317] [General]
Hello, 
a couple of days back I updated the rspack version from 0.1.8 to 0.1.12 and I started getting errors when parsing map files from 3rd party dependencies. 

error[javascript]: JavaScript parsing error
  ┌─ ..\..\..\node_modules\@msstream\components-web-recorder\loc\language-tr.js.map:1:11
  │
1 │ {"version":3,"file":"../loc/language-tr.js","mappings":"CAAA,SAA2CA,EAAM

I was not getting this error with the previous 0.1.8 version and I am not sure why with the upgrade I would get this error. Any ideas, help to get around this issue? 

Thanks

**Comment by IWANABETHATGUY** [comment]
Could you provide more context, like a minimum reproduce repo, or you configuration? It seems rspack recognizes the `.js.map` as a javascript file, can you double check your `test` regex in your rules configuration? Make sure you don't write some `rule` like
```js
module.exports = {
// ...
  module: {
		rules: [
			{
				test: /\.js/
			}
		]
	}

// ...
}
```

the **correct** one should be 
```diff
module.exports = {
// ...
  module: {
		rules: [
			{
-				test: /\.js/
+				test: /\.js$/
			}
		]
	}

// ...
}
```

**Comment by pushkargupta9807** [comment]
Thank you for your response - all the rules defined with .js files have a terminating '$' . For instance,
        {
          test: /\.m?js$/,
          include: /node_modules/,
          type: "javascript/auto",
          resolve: {
            preferRelative: true,
            fullySpecified: false,
          },
        },
                {
          test: /[\/\\]@ms[\/\\].+[\/\\]lib[\/\\].+\.js$/,
          use: {
            loader: "@ms/items-view-module-loader",
          },
        },
         {
          test: /\.resx.js$/,
}

The weird part is 
1) until 2 weeks back when I was on 0.1.8, I was not getting this error with rspack. 
2) I am switching back and forth between webpack and rspack using the almost the same configuration - and with webpack I do not get this error.

**Comment by pushkargupta9807** [comment]
Also is there a rule I can use to "ignore" all the *.map.js files in node_modules ? 
For instance, something like :
                 {
                    test: /\.js.map$/,
                    include: /node_modules/
                    action: "ignore",

                  },

**Reply by IWANABETHATGUY** [reply]
Can you give me a minimal reproduce repository link?

**Reply by pushkargupta9807** [reply]
I got around that issue now with using a "ignore-plugin" for all those map files. I can revisit this and upload the repository soon. I guess the question remains why the issue started showing with upgrading from 0.1.8 to 0.1.12. Thanks!

**Reply by IWANABETHATGUY** [reply]
Yeah, there are too many commits between 0.0.8 and 0.0.12, so a repo would be very helpful
